packages/example-app-base/src/utilities/extractAnchorLink.ts

```diff
@@ -1,10 +1,15 @@
 /**
- * Returns the last hash value in the link
+ * Returns the last hash value in `url`.
+ * If `url` doesn't contain a hash, returns an empty string.
+ * @param url - Full or partial URL. Just the hash section is valid, as long as it's prepended with #.
+ * It can also contain multiple hashes, the first for the route and the second for an anchor on the page
+ * (like `#/components/checkbox#Overview`).
  */
-export function extractAnchorLink(url: string): string | undefined {
-  const split = url.split('#');
-  const cleanedSplit = split.filter((value: string) => !!value);
-  if (cleanedSplit.length > 1) {
+export function extractAnchorLink(url: string): string {
+  if (url.indexOf('#') !== -1) {
+    const split = url.split('#');
+    const cleanedSplit = split.filter((value: string) => !!value);
     return cleanedSplit[cleanedSplit.length - 1];
```

Comment:
if there is only 1 hash, is that value always the anchor?

Reply:
Thanks for having JSDoc btw but either the JSDoc could use more elaboration or it could use tests

Reply:
This is for the very specific case of the example app, where the URLs have a first hash for the route and possibly a second hash for the anchor (which is why we need all kinds of hacks for scrolling to anchors). While trying to explain all that in documentation (even before seeing your comment) I also decided it needed a test. :P